src/crystal/system/unix/process.cr

```diff
@@ -0,0 +1,254 @@
+require "c/signal"
+require "c/stdlib"
+require "c/sys/resource"
+require "c/unistd"
+
+class Process
+  protected def wait_system
+    Crystal::SignalChildHandler.wait(@waitpid, @pid)
+  end
+
+  protected def self.exit_system(status = 0) : NoReturn
+    LibC.exit(status)
+  end
+
+  protected def self.pid_system : Int64
+    LibC.getpid.to_i64
+  end
+
+  protected def self.ppid_system : Int64
+    LibC.getppid.to_i64
+  end
+
+  protected def exists_system?
+    Process.exists_system?(@pid)
+  end
+
+  protected def self.exists_system?(pid : Int64)
+    ret = LibC.kill(pid, 0)
+    return true if ret == 0
+    return false if Errno.value == Errno::ESRCH
+    raise Errno.new("kill")
+  end
+
+  protected def pgid_system : Int64
+    Process.pgid_system(@pid)
+  end
+
+  protected def self.pgid_system(pid : Int64) : Int64
+    ret = LibC.getpgid(pid)
+    raise Errno.new("getpgid") if ret < 0
+    ret.to_i64
+  end
+
+  protected def signal_system(signal : Signal)
+    ret = LibC.kill(@pid, signal.value)
+    raise Errno.new("kill") if ret < 0
+  end
+
+  protected def create_and_exec(command : String, args : (Array | Tuple)?, env : Env?, clear_env : Bool, fork_input : IO::FileDescriptor, fork_output : IO::FileDescriptor, fork_error : IO::FileDescriptor, chdir : String?)
```

Comment:
Perhaps call this just `spawn`?

Reply:
I prefer the explicit name than a confusing name with spawning a fiber.